blog/2024-12-15-openhab-4-3-release.md

```diff
@@ -0,0 +1,52 @@
+---
+layout: BlogPost
+category: blog
+title: openHAB 4.3 Release
+author: Florian Hotze and others
+date: '2024-12-15T18:00:00+01:00'
+previewimage: /uploads/2024-12-15-openhab-4-3-release/header.png
+tags:
+  - news
+excerpt: >-
+
+---
+
+# Activity
+
+# Highlights
+
+As you might have already noticed in the activity statistics, there was much development on openHAB - too much to list everything.
+Please refer to the [**release notes**](https://github.com/openhab/openhab-distro/releases/tag/4.3.0) to check out what's new and noteworthy (e.g. breaking changes).
+
+In the following sections, our maintainers and contributors are introducing some new features you don't want to miss!
+
+## Core Runtime Enhancements
+
+## Main UI Enhancements
+
+_Florian Hotze ([@florian-h05](https://github.com/florian-h05)), openHAB Maintainer_
+
+## Blockly Enhancements
+
+_Stefan Höhn ([@stefan-hoehn](https://github.com/stefan-hoehn)), Blockly Maintainer_
+
+## Sitemap Enhancements
+
+_Laurent Garnier ([@lolodomo](https://github.com/lolodomo)), openHAB Maintainer_
+
+## Basic UI Enhancements
+
+_Laurent Garnier ([@lolodomo](https://github.com/lolodomo)), openHAB Maintainer_
+
+## Add-on Enhancements
+
+openHAB 4.3 brings XX new add-ons, adding support for new devices:
+
+But not only were new bindings added, some existing bindings also received massive improvements:
+
+## openHABian Enhancements
+
```

Comment:
```suggestion

# An outlook towards openHAB 5

openHAB 4.3 will be the last major release of the openHAB 4 series, and the last major release which supports 32-bit systems.
With openHAB 5.0, our core engine will require Java 21, compared to Java 17 used in the 4.x series. 
Due to the limited availability of some library dependencies and Java 21 distributions for 32-bit systems, we also decided to focus on 64-bit systems for openHAB 5.x.
To be prepared for the next major release, you will require a 64-bit operating system on your openHAB server.
Just keep this in mind if you think about reinstalling your openHAB installation.

```

Reply:
Regarding the last line, it is not only something to consider when re installing but also when you pick hardware (RPI4 or later), might be good to add.

Reply:
I did not want to be too technical here, as this is the blog post.... But if you think it helps, we can also extend this a bit.

But we should revisit our hardware recommendations on the web page already now.

Reply:
> Regarding the last line, it is not only something to consider when re installing but also when you pick hardware (RPI4 or later), might be good to add.

Disagree, any recent HW can do 64. Available since RPi3.
Just omit references to HW and instead spend some more words to motivate selecting 64 bit from NOW on whenever they choose to reinstall, be it as part of their migration to 4.3 or in the meantime until OH5 gets out.
 
> this also might not be the most popular upcoming change, so maybe we can spend 1 or 2 sentences to the ‘why’ (Java 17 eol, security, dependencies)

I'd omit the half-hearted explication that it's due to libraries.

The real why is none of the above but that it's the right time and opportunity to make the cut NOW.
And that's how we should announce and argument it.
People with non-64 HW can stay with OH4. Anyone else can seize the opportunity and reinstall/lift his OS to 64, too.

Reply:
@holgerfriedrich Yeah, maybe not be to technical. But without any reason, I expect it will lead to questions and discussion.

@mstormi I was talking about why we need to upgrade, not about why we do it now. Do you think it is important to tell why we upgrade now? 

Regarding the hardware I wrote rpi4 but I was wrong and meant rpi3. Looking at https://www.openhab.org/docs/installation/openhabian.html#hardware-recommendation we have already adapted the documentation in sync with the other GitHub discussion.

Reply:
> not about why we do it now. Do you think it is important to tell why we upgrade now?

Yes I do. When people will be confronted with this decision, they don't have a vote.
Image you dislike it - would you want to be told by those that took the decision that they rate reasons other than yours higher? In the end, what the reasons are doesn't matter to them when they cannot change it.
The 'clean cut' and 'enough time to prepare' storytelling on the other hand is pretty sound for everybody.

Reply:
Are you happy with the currently suggested text?

Reply:
I issued a PR.
Now the OH5 outlook part is still missing in your repo fork proposal, isn't it ?
Please point us at the current proposal for that paragraph.

Reply:
I meant the proposal that @holgerfriedrich posted at the beginning of this conversation (https://github.com/openhab/website/pull/499#discussion_r1878804000), but looking at its history I guess it hasn‘t been adjusted after your conversation here.

Reply:
@florian-h05 I have not updated the text, an actually I like @mstormi's motivation part (first half of the paragraph). But I'm not sure if we should recommend moving to 64-bit NOW.
I think one major piece is missing for that - an actual documentation (or tool extension) to backup/restore the persistence databases to the new installation. This has also been mentioned in the planning for OH5. The available backup/restore you are referring to only cares for OH settings. That's why I tried to put it more like "if you (re)install anyway, pick 64-bit".